### PR TITLE
Implement dump_smt2 for MathSAT backend

### PR DESCRIPTION
**Pull request by makaimann**

### CODE REVIEW

File: msat/src/msat_solver.cpp [hunk 1]
```diff
@@ -846,7 +847,22 @@ Term MsatSolver::substitute(const Term term,
 
 void MsatSolver::dump_smt2(FILE * file) const
 {
-  throw NotImplementedException("Can't dump all assertions to a file yet");
+  size_t num_asserted;
+  msat_term * assertions = msat_get_asserted_formulas(env, &num_asserted);
+  msat_term * ait = assertions;
```
Comment (ahmed-irfan) [inline msat/src/msat_solver.cpp]: why do we need this extra pointer

File: msat/src/msat_solver.cpp [hunk 2]
```diff
@@ -846,7 +847,22 @@ Term MsatSolver::substitute(const Term term,
...
+  msat_term all_asserts = msat_make_true(env);
+  for (size_t i = 0; i < num_asserted; i++)
```
Comment (ahmed-irfan) [inline msat/src/msat_solver.cpp]: ++i

File: msat/src/msat_solver.cpp [hunk 3]
```diff
@@ -846,7 +847,22 @@ Term MsatSolver::substitute(const Term term,
...
+  {
+    all_asserts = msat_make_and(env, all_asserts, *ait);
+    ait++;
```
Comment (ahmed-irfan) [inline msat/src/msat_solver.cpp]: ++ait

File: msat/src/msat_solver.cpp (hunk 1 above)
Comment (makaimann) [reply msat/src/msat_solver.cpp]: Oh yeah, good catch. That was left over from something else I was trying. Fixed with 75bb8e4bc5d592bc446f0f4debfa68caf44ada46

**Pull request merged**